packages/@aws-cdk/aws-appmesh/lib/virtual-node-listener.ts

```diff
@@ -54,6 +57,32 @@ export interface HttpVirtualNodeListenerOptions extends VirtualNodeListenerCommo
    * @default - None
    */
   readonly timeout?: HttpTimeout;
+
+  /**
+   * Connection pool for http listeners
+   *
+   * @default - None
+   */
+  readonly connectionPool?: HttpConnectionPool;
+}
+
+/**
+ * Represent the HTTP2 Node Listener prorperty
+ */
+export interface Http2VirtualNodeListenerOptions extends VirtualNodeListenerCommonOptions {
+  /**
+   * Timeout for HTTP protocol
+   *
+   * @default - None
+   */
+  readonly timeout?: HttpTimeout;
```

Comment:
Same note here - let's extract a common, module-private interface that extends `VirtualNodeListenerCommonOptions` and contains the `timeout` property, which both `HttpVirtualNodeListenerOptions` and `Http2VirtualNodeListenerOptions` can extend.

Reply:
They already all extend `VirtualNodeListenerCommonOptions`, the timeout field is one of the ones that differs by protocol. `Http` and `Http2` are the only options out of 4 that share the same timeout so I'd like to leave this one.

Reply:
Why...?